beets/dbcore/types.py

```diff
@@ -168,52 +209,66 @@ def format(self, value: int) -> str:
         return '{}{}'.format((value or 0) // self.unit, self.suffix)
 
 
-class Id(Integer):
+class Id(NullInteger):
     """An integer used as the row id or a foreign key in a SQLite table.
     This type is nullable: None values are not translated to zero.
     """
-    null = None
+    @property
+    def null(self) -> None:
+        return None
 
     def __init__(self, primary: bool = True):
         if primary:
             self.sql = 'INTEGER PRIMARY KEY'
 
 
-class Float(Type):
+class BaseFloat(Type[float, N]):
     """A basic floating-point type. The `digits` parameter specifies how
     many decimal places to use in the human-readable representation.
     """
     sql = 'REAL'
-    query = query.NumericQuery
+    query = NumericQuery
     model_type = float
 
     def __init__(self, digits: int = 1):
         self.digits = digits
 
-    def format(self, value: float) -> str:
+    def format(self, value: Union[float, N]) -> str:
         return '{0:.{1}f}'.format(value or 0, self.digits)
 
 
-class NullFloat(Float):
+class Float(BaseFloat[float]):
+    pass
```

Comment:
Nice; this is a particularly good idea to make a new base class here. You could imagine leaving `null` unimplemented in the base, which would more closely match `BaseInteger`, but it doesn't seem to matter too much.

Reply:
Done! Matching `Float` and `Integer` makes a lot of sense here.